torch/csrc/distributed/c10d/ProcessGroup.hpp

```diff
@@ -23,6 +24,16 @@ constexpr auto kProcessGroupDefaultTimeout =
 
 namespace c10d {
 
+C10_EXPORT void register_work(
+    const at::Tensor& tensor,
+    const c10::intrusive_ptr<c10d::Work>& work);
+
+C10_EXPORT at::Tensor wait_tensor(const at::Tensor& tensor);
+
+C10_EXPORT void unregister_completed_works();
+
+C10_EXPORT size_t get_work_registry_size();
```

Comment:
Are these two functions only used for testing purposes?

Reply:
yes - `get_work_registry_size` is only for testing purposes